### DISCUSSION TITLE
After automatic update without backup beforehand database container fails to start

### DISCUSSION THREAD

**Discussion by NiklasReisser** [nextcloud/all-in-one#5047] [Questions] [answered]
Hi, 

after updating my nextcloud instance, I am unable to bring it back online again. 
Thanks in advance for any help & support!

Kind regards

Niklas

### Steps to reproduce
1. I followed https://github.com/nextcloud/all-in-one#how-to-enable-automatic-updates-without-creating-a-backup-beforehand

### Expected behavior 

nextcloud should be up and running again

### Actual behavior 

The apache, nextcloud, database and notify-push containers remain in state "starting"

The cause the database container being stuck in a reboot loop, with the log below. The other containers are "just" waiting for the database

```
2024-07-26T09:26:32.651127026Z Unable to restore the database because the database dump is missing.
2024-07-26T09:26:33.197662313Z Unable to restore the database because the database dump is missing.
2024-07-26T09:26:33.665916939Z Unable to restore the database because the database dump is missing.
2024-07-26T09:33:19.988073491Z Unable to restore the database because the database dump is missing.
... <continues endlessly> ...
```

### Host OS 

docker / ubuntu 23.10

#### Nextcloud AIO version 

v9.3.0 (after the update. Unfortunately I don't recall the previous version)

#### Current channel 

latest

#### Other valuable info 

1. I followed this instructions because I have a separate backup solution in place.
2. nextcloud-apache is not avalailable at the assigned port (11000) as well. Stopping the containers only works via AIO-WebGui, calling `docker exec --env STOP_CONTAINERS=1 nextcloud-aio-mastercontainer /daily-backup.sh` causes endless `Waiting for apache to become available` messages

**Comment by szaimen** [comment]
Hi, please restore a backup from before the update to recover from this situation.

**Reply by NiklasReisser** [reply]
Hi, thanks for the reply. 

But why is this not a bug? And, since as far I am can tell, I did everything according to the documentation, how can I avoid this error in the future?

**Reply by szaimen** [reply]
It is not expected that no database dump was created. Which OS is the host running?

**Reply by NiklasReisser** [reply]
I am using Ubuntu 23.10

**Reply by szaimen** [reply, marked as answer]
Ah I see. Then you probably ran into this: https://github.com/nextcloud/all-in-one/discussions/4390#discussioncomment-8800926

**Reply by szaimen** [reply]
Regarding 24.04 LTS compatibility, it is currently planned with https://github.com/nextcloud/all-in-one/issues/4775

**Reply by NiklasReisser** [reply]
Yes, that makes sense. The point I get the error is probably different due to the different backup solution?! Its however odd that none of my other containers have this problem. I'll try to have a look once I get my instance back up.

Unortunately it looks like the problem was around for a while, but I didnt notice it. So my backups don't contain a dump either. :(
I had a look at the code, if I understand it correctly the container only tries to import the dump due to an update of the postgres version? 
And I still have the content of nextcloud-aio-database. So there should be a way to create the dump manually? Something like starting the old postgres version, call the correct commands from start.sh?

I am happy to figure out the commands myself, but is my understanding generally correct?

**Reply by szaimen** [reply]
I think you might be able to fix this by downgrading to `nextcloud/all-in-one:20240606_075829-latest`, restoring the backup and then doing a database dump manually. Afterwards try to upgrade to normal `latest` channel

**Reply by NiklasReisser** [reply]
Yes, that worked. Thanks for the support! 

For anyone else stumbling across this: 

There are workarounds for the docker stop problem described in the upstream issues, e.g. https://bugs.launchpad.net/ubuntu/+source/apparmor/+bug/2039294

The relevant commands & values for the manual export can be found here: 
https://github.com/nextcloud/all-in-one/blob/89d2dd7287c19597b6a3c694b0648aea60bf1b16/Containers/postgresql/start.sh#L175
